src/wallet/wallet.cpp

```diff
@@ -2797,7 +2798,7 @@ bool CWallet::CreateTransactionInternal(
         txNew.nLockTime = GetLocktimeForNewTransaction(chain(), GetLastBlockHash(), GetLastBlockHeight());
         {
             std::vector<COutput> vAvailableCoins;
-            AvailableCoins(vAvailableCoins, true, &coin_control, 1, MAX_MONEY, MAX_MONEY, 0);
+            AvailableCoins(vAvailableCoins, !coin_control.m_include_unsafe_inputs, &coin_control, 1, MAX_MONEY, MAX_MONEY, 0);
```

Comment:
We could remove the `fOnlySafe` argument to `AvailableCoins` now that this is provided by `coin_control`.

The only gotcha is that `AvailableCoins` can be called without a `coin_control`, but I checked the codebase and in all the places where that happens `fOnlySafe` is set to `true`, so we could set it inside `AvailableCoins` to `const bool only_safe = {coinControl ? !coinControl->m_include_unsafe_inputs : true};`.

Do you want me to apply that change in this PR?